### PR TITLE
Feature/ccd 0052 admin accounts

### PR DESCRIPTION
**Pull request by ShawnWang86b**

### CODE REVIEW

File: src/admin/admin.service.ts [hunk 1]
```diff
@@ -99,12 +104,64 @@ export class AdminService {
...
+      ...UpdateAdminDto,
       hashedRefreshToken: hashedRefreshToken,
       updatedAt: new Date(),
     };
     await this.adminModel
       .findByIdAndUpdate({ _id: adminId }, { $set: updateAdminDto }, { new: true })
       .exec();
   }
+
+  async findAll(paginationQuery: PaginationQueryDto) {
```
Comment (RedRe4per) [inline src/admin/admin.service.ts:116]: : Promise<Admin[]>

File: src/admin/dto/update-admin.dto.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  @IsNotEmpty()
+  @IsString()
+  readonly permission: string;
+
+  readonly isDeleted: boolean;
+
+  @IsNotEmpty()
+  @IsString()
+  readonly name: string;
+}
```
Comment (RedRe4per) [inline src/admin/dto/update-admin.dto.ts:25]: A small suggestion. I saw the update admin dto is almost a copy of create admin dto. I believe you can try to use partial type.

`export class UpdateAdminDto extends PartialType(CreateAdminDto) {
  xxxxx // your changes
}`

File: src/admin/dto/create-admin.dto.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  readonly hashedRefreshToken: string;
+
+  readonly createdAt: Date;
+
+  readonly updatedAt: Date;
+
+  @IsNotEmpty()
+  readonly permission: string;
+
+  readonly isDeleted: boolean;
```
Comment (RedRe4per) [inline src/admin/dto/create-admin.dto.ts:21]: I believe you can simply delete `readonly isDeleted: boolean;` . When you create a new admin account, it must be isDeleted: false.

File: migrations/20221016094040-update-add-permission-to-admin.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  async up(db) {
+    await db.collection("admins").updateMany({}, { $set: { permission: "" } });
+
+    await db
+      .collection("admins")
+      .updateOne({ email: "court.canva@gmail.com" }, { $set: { permission: "Super" } });
+  },
+
```
Comment (RedRe4per) [inline migrations/20221016094040-update-add-permission-to-admin.js:9]: if you add `@Prop() name: string;` in schema and set it `@IsNotEmpty()` in dto, please give admin `email: "court.canva@gmail.com"` a name here.

**Review by RedRe4per — CHANGES_REQUESTED** [review]

File: src/admin/admin.controller.ts [hunk 5]
```diff
@@ -1,24 +1,70 @@
...
+import { UpdateAdminDto } from "./dto/update-admin.dto";
+import { LoginAdminDto } from "./dto/login-admin.dto";
+import { PaginationQueryDto } from "../utils/PaginationDto/pagination-query.dto";
 
 @Controller("admin")
 export class AdminController {
   constructor(private adminService: AdminService) {}
 
+  @Get()
+  async findAllAdmin(@Query() paginationQuery: PaginationQueryDto) {
```
[the remark this answers is not in the document]
Comment (ShawnWang86b) [reply src/admin/admin.controller.ts:31]: will add

File: src/admin/admin.service.ts (hunk 1 above)
Comment (ShawnWang86b) [reply src/admin/admin.service.ts:116]: will add

File: src/admin/dto/update-admin.dto.ts (hunk 2 above)
Comment (ShawnWang86b) [reply src/admin/dto/update-admin.dto.ts:25]: thx modified it to PartialType

`import { PartialType } from "@nestjs/mapped-types";
import { CreateAdminDto } from "./create-admin.dto";

export class UpdateAdminDto extends PartialType(CreateAdminDto) {}
`

File: migrations/20221016094040-update-add-permission-to-admin.js (hunk 4 above)
Comment (ShawnWang86b) [reply migrations/20221016094040-update-add-permission-to-admin.js:9]: thx changed the migration file;

module.exports = {
  async up(db) {
    await db
      .collection("admins")
      .updateOne(
        { email: "court.canva@gmail.com" },
        { $set: { permission: "Super", name: "Canva" } },
      );
  },

  async down(db) {
    await db
      .collection("admins")
      .updateOne({ email: "court.canva@gmail.com" }, { $unset: { permission: null, name: null } });
  },
};

File: src/admin/dto/create-admin.dto.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  @IsNotEmpty()
+  @IsString()
+  readonly password: string;
+
+  readonly hashedRefreshToken: string;
+
+  readonly createdAt: Date;
+
+  readonly updatedAt: Date;
```
[the remark this answers is not in the document]
Comment (ShawnWang86b) [reply src/admin/dto/create-admin.dto.ts:16]: delete the createAt and updateAt from **create-admin. dto** and **update-admin.dto**

File: src/admin/dto/create-admin.dto.ts (hunk 3 above)
Comment (ShawnWang86b) [reply src/admin/dto/create-admin.dto.ts:21]: deleted

File: src/admin/schemas/admin.schema.ts [hunk 7]
```diff
@@ -20,6 +20,12 @@ export class Admin extends Document {
 
   @Prop({ default: false })
   isDeleted: boolean;
+
+  @Prop()
+  permission: string;
```
[the remark this answers is not in the document]
Comment (ShawnWang86b) [reply src/admin/schemas/admin.schema.ts:25]: thx but met some block, let's have a catch-up when you are free

**Review by RedRe4per — APPROVED** [review]
LGTM

**Review by GLC-coder — APPROVED** [review]
LGTM

**Review by PengchengYuGithub — DISMISSED** [review]

**Pull request merged**